middlewares/accesslog/logger.go

```diff
@@ -23,11 +29,34 @@ const (
 // Note: Current implementation collects log data but does not have the facility to
 // write anywhere.
 type LogHandler struct {
+	logger *logrus.Logger
+	file   *os.File
 }
 
 // NewLogHandler creates a new LogHandler
-func NewLogHandler() *LogHandler {
-	return &LogHandler{}
+func NewLogHandler(fileName string) (*LogHandler, error) {
+	if len(fileName) > 0 {
+		dir := filepath.Dir(fileName)
+
+		err := os.MkdirAll(dir, 0755)
+		if err != nil {
+			log.Errorf("Failed to create log path %s: %s", dir, err)
+		}
+
+		file, err := os.OpenFile(fileName, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0664)
+		if err != nil {
+			log.Error("Error opening file", err)
+		}
+		logger := &logrus.Logger{
+			Out:       file,
+			Formatter: new(CommonLogFormatter),
+			Hooks:     make(logrus.LevelHooks),
+			Level:     logrus.InfoLevel,
+		}
+		return &LogHandler{logger: logger, file: file}, nil
+	}
+
+	return nil, errors.New("Empty file name specified for accessLogsFile")
```

Comment:
Please move this return at the very top of the file, right after a check for `fileName == ""`. That way, we can save one level of indentation.